rdtools/filtering.py

```diff
@@ -102,25 +185,325 @@ def clip_filter(power_ac, quantile=0.98):
+                      roll_periods=None):
     '''
+    This filter is a logic-based filter that is used to filter out
+    clipping periods in AC power and AC energy time series.
+    The AC power or energy time series is filtered based on the
+    maximum derivative over a rolling window, as compared to a user-set
+    derivate_cutoff (default set to 0.2).  The size of the
+    rolling window is increased when the system is a tracked system.
+    
+    Parameters
+    ----------
+    power_ac : pd.Series
+        Pandas time series, representing PV system power or energy with
+        a pandas datetime index.
+    mounting_type: string, default 'Fixed'
+        String representing the mounting configuration associated with the
+        AC power/energy time series. Can either be "Fixed" or "Tracking".
+        Default set to 'Fixed'.
+    derivative_cutoff : float, default 0.2
+        Cutoff for max derivative threshold. Defaults to 0.2; however, values
+        as high as 0.4 have been tested and shown to be effective.  The higher
+        the cutoff, the more values in the dataset that will be determined as
+        clipping.
+    roll_periods: Integer.
+        Number of periods to examine when looking for a near-zero derivative
+        in the time series derivative. If roll_periods = 3, the system looks
+        for a near-zero derivative over 3 consecutive readings. Default value
+        is set to None, so the function uses default logic: it looks for a
+        near-zero derivative over 3 periods for a fixed tilt system, and over
+        5 periods for a tracked system with a sampling frequency more frequent
+        than once every 30 minutes.
 
-    csi = poa_global_measured / poa_global_clearsky
-    return (csi >= 1.0 - threshold) & (csi <= 1.0 + threshold)
+    Returns
+    -------
+    pd.Series
+        Boolean Series of whether to include the point because it is not
+        clipping.
+        True values delineate non-clipping periods, and False values delineate
+        clipping periods.
+    '''
+    # Format the power time series
+    power_ac, index_name = _format_clipping_time_series(power_ac, 
+                                                        mounting_type)
+    # Get the sampling frequency of the time series
+    time_series_sampling_frequency = power_ac.index.to_series().diff()\
+        .astype('timedelta64[m]').mode()[0]
+    # Make copies of the original inputs for the cases that the data is
+    # changes for clipping evaluation
+    original_time_series_sampling_frequency = time_series_sampling_frequency
+    power_copy = power_ac.copy()
+    # Drop duplciate indices
+    power_ac = power_ac.reset_index().drop_duplicates(
+                        subset=power_ac.index.name,
+                        keep='first').set_index(power_ac.index.name)
+    freq_string = str(time_series_sampling_frequency) + 'T'
+    if time_series_sampling_frequency >= 10:
+        power_ac = power_ac.asfreq(freq_string)
```

Comment:
For conforming to a regular frequency, what do you think about interpolating instead of just using `asfreq()`?